src/Components/Components/src/ComponentBase.cs

```diff
@@ -175,22 +190,22 @@ void IComponent.Configure(RenderHandle renderHandle)
         /// Method invoked to apply initial or updated parameters to the component.
         /// </summary>
         /// <param name="parameters">The parameters to apply.</param>
-        public virtual Task SetParametersAsync(ParameterCollection parameters)
+        Task IComponent.SetParametersAsync(ParameterCollection parameters)
         {
-            parameters.SetParameterProperties(this);
```

Comment:
This was another issue I discovered here (or maybe it's intentional and I just missed it).

The way this is currently implemented, Init runs **after** parameters are set. This means that you can read parameter values in Init and it works. That seems really surprising and I wonder how often it's misused. I assumed that it's valuable to separate these things so that we can keep the exact timing of init being called an implementation detail.